File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        if nums == None or len(nums) < 2:
            return False
        
        last_seen = set()
        for num in nums:
            if num in last_seen:
                return True
            last_seen.add(num)

        return False

File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> allPossibleSubsets = new ArrayList<>();
        List<Integer> current = new ArrayList();
        int index = 0;

        this.backTracking(index, nums, current, allPossibleSubsets);
        
        return allPossibleSubsets;
    }

    private void backTracking(int index, int[] nums, List<Integer> current, List<List<Integer>> allPossibleSubsets) {
        if (index == nums.length) {
            allPossibleSubsets.add(new ArrayList<>(current));
            return;
        }

        current.add(nums[index]);
        backTracking(index + 1, nums, current, allPossibleSubsets);
        current.remove(current.size() - 1);
        backTracking(index + 1, nums, current, allPossibleSubsets);

    }
}
